onnxruntime/core/providers/coreml/builders/impl/pad_op_builder.cc

```diff
@@ -0,0 +1,215 @@
+  // Add operator related
+#ifdef __APPLE__
+ public:
+  void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const override;
+
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override;
+#endif
+
+  // Operator support related
+ private:
+  bool IsOpSupportedImpl(const Node& node, const OpBuilderInputParams& input_params,
+                         const logging::Logger& logger) const override;
+
+  int GetMinSupportedOpSet(const Node& /* node */) const override {
+    // Note: before Pad-11, inputs `pads` and `constant_value` were attributes
+    return 11;
+  }
+};
+
+// Add operator related
+
+#ifdef __APPLE__
+void PadOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const {
+  model_builder.AddInitializerToSkip(node.InputDefs()[1]->Name());  //  pads
+  model_builder.AddInitializerToSkip(node.InputDefs()[2]->Name());  //  constant_value
+  if (node.InputDefs().size() > 3) {
+    model_builder.AddInitializerToSkip(node.InputDefs()[3]->Name());  // axes
+  }
+}
+
+Status PadOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                           const Node& node,
+                                           const logging::Logger& logger) const {
+  std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = CreateNNLayer(model_builder, node);
+
+  auto* coreml_pad = layer->mutable_padding();
+  auto* constant_padding_type = coreml_pad->mutable_constant();  // CoreML::Specification::PaddingLayerParams_PaddingConstant
+
+  const auto& input_defs = node.InputDefs();
+  std::vector<int64_t> input_shape;
+  GetShape(*input_defs[0], input_shape, logger);
+  const auto input_rank = input_shape.size();
+
+  const auto& pads_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());            // pads
+  const auto& constant_value_tensor = *model_builder.GetInitializerTensors().at(input_defs[2]->Name());  // constant_value
+
+  Initializer constant_value_raw_data_init(constant_value_tensor);
+  float constant_value = constant_value_raw_data_init.DataAsSpan<float>()[0];
+  constant_padding_type->set_value(constant_value);
+
+  Initializer pads_initializer_raw_data(pads_tensor);
+  auto pads_span = pads_initializer_raw_data.DataAsSpan<int64_t>();
+
+  // if not provided, make a default axes as [0, 1, ..., input_rank - 1]
+  std::vector<int64_t> default_axes(input_rank);
+  std::iota(std::begin(default_axes), std::end(default_axes), 0);
+  auto axes_tensor_data = gsl::span<const int64_t>(default_axes);
+  if (input_defs.size() > 3) {
+    // optional input axes is provided
+    const ONNX_NAMESPACE::TensorProto& axes_tensor = *model_builder.GetInitializerTensors().at(input_defs[3]->Name());
+    Initializer axes_initializer_raw_data(axes_tensor);
+    axes_tensor_data = axes_initializer_raw_data.DataAsSpan<int64_t>();
```

Comment:
after this scope `axes_initializer_raw_data` will be destroyed and I think `axes_tensor_data` will point to freed memory.
 `std::variant<Initializer, InlinedVector<int64_t>>` in the enclosing scope might make sense here.

Reply:
similar comment in PadOpBuilder::IsOpSupportedImpl